device-registry/src/test/scala/org/genivi/sota/device_registry/DeviceRequests.scala

```diff
@@ -42,8 +42,16 @@ trait DeviceRequests { self: ResourceSpec =>
   def fetchDevice(uuid: Uuid): HttpRequest =
     Get(Resource.uri(api, uuid.show))
 
-  def searchDevice(namespace: Namespace, regex: String): HttpRequest =
-    Get(Resource.uri(api).withQuery(Query("namespace" -> namespace.get, "regex" -> regex)))
+  def searchDevice(namespace: Namespace, regex: String, deviceId: Option[DeviceId] = None, offset: Long = 0,
+                   limit: Long = 50): HttpRequest =
+    deviceId match {
+      case Some(d) =>
+        Get(Resource.uri(api).withQuery(Query("deviceId" -> d.underlying, "regex" -> regex, "offset" -> offset.toString,
+          "limit" -> limit.toString, "namespace" -> namespace.get)))
+      case None =>
+        Get(Resource.uri(api).withQuery(Query("regex" -> regex, "offset" -> offset.toString, "limit" -> limit.toString,
+          "namespace" -> namespace.get)))
```

Comment:
Sorry, it seems I confused myself with the `deviceId`, the point I wanted to make is that the `namespace` query isn't used at all and isn't needed[1], and then I saw the `deviceId` field and got confused. In fact it is an error to provide both an `regex` and a `deviceId` and the method `fetchByDeviceId` is using the `deviceId` field correctly. 

Sorry for this, can you revert (possibly remove the namespace, or send it as a header in case we actually want to test namespaces in the future).

[1] This all works since the tests are only using the default namespace anyway